src/main/java/com/thegrizzlylabs/sardineandroid/impl/OkHttpSardine.java

```diff
@@ -349,12 +359,20 @@ public void move(String sourceUrl, String destinationUrl) throws IOException {
 
     @Override
     public void move(String sourceUrl, String destinationUrl, boolean overwrite) throws IOException {
-        Request request = new Request.Builder()
+        move(sourceUrl, destinationUrl, overwrite, null);
+    }
+
+    @Override
+    public void move(String sourceUrl, String destinationUrl, boolean overwrite, String lockToken) throws IOException {
+        Request.Builder buildsr = new Request.Builder()
                 .url(sourceUrl)
                 .method("MOVE", null)
                 .header("DESTINATION", URI.create(destinationUrl).toASCIIString())
-                .header("OVERWRITE", overwrite ? "T" : "F")
-                .build();
+                .header("OVERWRITE", overwrite ? "T" : "F");
+        if (lockToken != null) {
+            buildsr.header("If", "<" + destinationUrl + "> (<" + lockToken + ">)");
```

Comment:
Can you move this line to a separate method to avoid repeating it in the `move` and `put` methods?